Add Scarf based telemetry (#39510)
mistercrunch/Airflow: docs/apache-airflow/faq.rst

#. Set ``explicit_defaults_for_timestamp = 1`` under the ``mysqld`` section in your ``my.cnf`` file.
#. Restart the Mysql server.

Does Airflow collect any telemetry data?
----------------------------------------

Airflow integrates `Scarf <https://about.scarf.sh/>`__ to collect basic telemetry data during operation.
This data assists Airflow maintainers in better understanding how Airflow is used.
Insights gained from this telemetry are critical for prioritizing patches, minor releases, and

> security fixes. Additionally, this information supports key decisions related to the development Roadmap.

Reviewer: ```suggestion
security fixes. Additionally, this information supports key decisions related to the development roadmap.
```

Author: Updated

> Users can easily opt out of analytics in various ways documented

Reviewer: ```suggestion
Users can easily opt-out of analytics in various ways documented
```

`here <https://docs.scarf.sh/gateway/#do-not-track>`__ and by setting the :ref:`config:scarf_analytics__enabled` option

> to ``False``. Airflow also respects ``SCARF_ANALYTICS=false`` environment variable.

Reviewer: ```suggestion
to ``False``. Airflow also respects the ``SCARF_ANALYTICS=false`` environment variable.
```

Author: Updated

> Users can easily opt-out of analytics in various ways documented
> `here <https://docs.scarf.sh/gateway/#do-not-track>`__ and by setting the :ref:`config:scarf_analytics__enabled` option
> to ``False``. Airflow also respects the ``SCARF_ANALYTICS=false`` environment variable.

Reviewer: ```suggestion
Deployments can opt-out of analytics by setting the ``enabled`` option
to ``False``, or the ``SCARF_ANALYTICS=false`` environment variable.
Individual users can easily opt-out of analytics in various ways documented in the
`Scarf Do Not Track docs <https://docs.scarf.sh/gateway/#do-not-track>`__.
```

> to ``False``. Airflow also respects the ``SCARF_ANALYTICS=false`` environment variable.

Reviewer 2: Can you please add a list of what is collected as metrics in current scope? Else users would need to take a look to codebase. Would be also very good for transparency to list it here.

Author: Will add it to the FAQ section

Author: Updated

#. Set ``explicit_defaults_for_timestamp = 1`` under the ``mysqld`` section in your ``my.cnf`` file.
#. Restart the Mysql server.

> Does Airflow collect any telemetry data?

Reviewer 3: i think this is too easily confused with otel / stats

Reviewer 3: maybe better to articulate as "usage tracking" or "analytics" or "usage analytics" or... something

Author: It can be yes but the description should help clear things up

----------------------------------------

.. _airflow-telemetry-faq:

Airflow integrates `Scarf <https://about.scarf.sh/>`__ to collect basic telemetry data during operation.
This data assists Airflow maintainers in better understanding how Airflow is used.
Insights gained from this telemetry are critical for prioritizing patches, minor releases, and
security fixes. Additionally, this information supports key decisions related to the development road map.

> 

Reviewer 3: worth adding some info about where it goes and how they can view it?

Reviewer 3: what about "right to forget" i.e. is there a mechanism for getting already-tracked data removed?

Reviewer 4: Aren't such rights only for PII data collected?

Author: They can't view it, atleast not now, just PMCs

Deployments can opt-out of analytics by setting the :ref:`[telemetry_collection] enabled <config:telemetry_collection__enabled>`

> option to ``False``, or the ``SCARF_ANALYTICS=false`` environment variable.

Reviewer 2: +1 here:
```suggestion
option to ``False``, or the ``AIRFLOW__TELEMETRY_COLLECTION__ENABLED=False`` environment variable.
```

Author: This is the alternate option. I want to respect what Scarf has in [their docs](https://docs.scarf.sh/package-analytics/#configuration) to disable too

https://github.com/apache/airflow/blob/ab1855a8d3a367f5ce36bf25a72ffebfe31448f7/airflow/settings.py#L579-L583

Author: If you look at rendered docs it says you can disable it via the "Airflow config" or the Scarf env variable.

The Airflow config can be set via Env Var, airflow.cfg or a Custom Secrets backend: https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html

Reviewer 2: Ah, sorry, over-looked this line. Thanks for clarification.

Individual users can easily opt-out of analytics in various ways documented in the
`Scarf Do Not Track docs <https://docs.scarf.sh/gateway/#do-not-track>`__.

The telemetry data collected is limited to the following:

- Airflow version
- Python version
- Operating system & machine architecture
- Executor

> - Metadata DB type & their version

Reviewer 5: nit: 
```suggestion
- Metadata DB type & its version
```

Author: fixed in https://github.com/apache/airflow/pull/39510/commits/93c94e6b48a93787654ff4c712681ed3945060fe
